charts/linkerd-control-plane/templates/destination.yaml

```diff
@@ -208,10 +208,15 @@ spec:
         {{- range (.Values.destinationController).experimentalArgs }}
         - {{ . }}
         {{- end }}
-        {{- with (.Values.destinationController).experimentalEnv }}
```

Comment:
tioli, I think we could minimize this with something like
```
        {{- with .Values.destinationController | concat .experimentalEnv .additionalEnv }}
```

I haven't tested this specific syntax but the general idea should avoid duplication.  Here and throughout.  Furthermore, we could consider using a partial here to reduce boilerplate